activation/poet.go

```diff
@@ -84,7 +84,13 @@ func NewHTTPPoetHarness(ctx context.Context, poetdir string, opts ...HTTPPoetOpt
 		return nil, err
 	}
 
-	client, err := NewHTTPPoetClient(cfg.RawRESTListener, PoetConfig{
+	// NewHTTPPoetClient takes an URL as connection string. Server speaks HTTP.
+	url := &url.URL{
+		Scheme: "http",
+		Host:   poet.GrpcRestProxyAddr().String(),
```

Comment:
i am confused.... isn't the whole point being able to configure the scheme?

Reply:
The `HTTPPoetClient` can take any URL now and does not assume `http` any more. 

The server we are starting with `NewHTTPHarness` exposes an address. Since we are connecting via localhost and are not using `https` in tests we can hardcode the scheme here.

The code here is only used in tests. The node can still be configured to connect to other poets via `https` now.

Reply:
oh! this is super confusing. can you move this function to a test then?

Reply:
I moved the `HTTPPoetHarness` to its own file and renamed it to `HTTPPoetTestHarness` with a clarifying comment that it is only to be used in tests.

@poszu maybe it makes sense to move `HTTPPoetClient` and `HTTPPoetTestHarness` into the poet repository? Maybe to `poet/client` and `poet/test` respectively? The only thing from go-spacemesh that the two depend on is the type `type.PoetServiceID` which is just an alias for `[]byte`.

Reply:
I created a ticket in the poet repository: https://github.com/spacemeshos/poet/issues/253

There are 2 implementations of the PoetHarness. I think it would be beneficial to merge those two into the poet repository and only use the harness here, instead of also having the implementation in `go-spacemesh`.